cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/DefaultEmitter.java

```diff
@@ -16,54 +16,64 @@
 
 package co.cask.cdap.etl.common;
 
+import co.cask.cdap.api.metrics.Metrics;
 import co.cask.cdap.etl.api.Emitter;
 import co.cask.cdap.etl.api.InvalidEntry;
 import co.cask.cdap.etl.api.StageMetrics;
 
 import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Iterator;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
  * Default implementation of {@link Emitter}. Tracks how many records were emitted.
  *
  * @param <T> the type of object to emit
  */
-public class DefaultEmitter<T> implements Emitter<T>, Iterable<T> {
-  private final List<T> entryList;
-  private final List<InvalidEntry<T>> errorList;
-  private final StageMetrics metrics;
+public class DefaultEmitter<T> implements Emitter<T> {
+  private final Map<String, List<T>> entriesMap;
```

Comment:
ok i see... this is being changed so that it is used throughout the whole pipeline instead of for a single stage. This was not how the Emitter was designed. We do not want a transform to specify which stage it is when emitting objects.